client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -141,7 +141,7 @@
      * @return A future which will throw if the operation fails, otherwise returning a boolean to
      *         indicate that the subscriber was updated in Stream Metadata.
      */
-    CompletableFuture<Boolean> updateReaderGroup(final String scopeName, final String rgName, ReaderGroupConfig config);
+    CompletableFuture<Long> updateReaderGroup(final String scopeName, final String rgName, ReaderGroupConfig config);
```

Comment:
IMO, this should return `updateReaderGroupResponse` instead of only returning Long -  generation. Since the return status is lost when we return only generation.